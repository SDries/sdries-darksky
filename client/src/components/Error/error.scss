@import '../../assets/styles/shared.scss';
@import '../../assets/styles/mixins.scss';

.Error {
  border: 1px solid $red;
  @include border-radius(4px);
  color: $red;
  padding: 20px;
  margin-bottom: 20px;
  font-size: 1.1rem;
  line-height: 1.1rem;

  p {
    font-weight: 700;
    margin-bottom: 10px;
  }

  button {
    background: $red;
    border: 1px solid $red;
    color: $white;
    font-size: .9rem;
    @include border-radius(4px);
    padding: 4px 10px;
    cursor: pointer;
    margin:0 auto;
    display:block;

    &:hover {
      background: $white;
      color: $red;
    }
  }
}
