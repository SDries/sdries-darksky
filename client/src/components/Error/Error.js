import React, { Component } from 'react';
import propTypes from 'prop-types';

import './error.scss'

class Error extends Component {
	constructor(props) {
		super(props);

		this.handleButtonClick = this.handleButtonClick.bind(this);
	}

	handleButtonClick() {
		if(this.props.buttonDisabled) return;
		this.props.buttonOnClick();
	}

	render(){
		let button;
		if(this.props.buttonOnClick) {
			button = <button
			onClick={this.handleButtonClick}
			disabled={this.props.buttonDisabled}
			>{this.props.buttonLabel}</button>;
		}

		return (
			<div className="Error">
			<p>{this.props.message}</p>
			{button}
			</div>
			);
	}
}

Error.propTypes = {
	message: propTypes.string.isRequired,
	buttonOnClick: propTypes.func
};

Error.defaultProps = {
	buttonDisabled: false,
	buttonLabel: 'Click here to retry',
};

export default Error;
