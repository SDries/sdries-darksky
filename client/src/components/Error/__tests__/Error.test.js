import React from 'react';
import Error from '../Error';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('react-error', () => {
  test('renders with message', () => {
    const tree = renderer.create(<Error message="message"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with button', () => {
    const mockCallBack = jest.fn();
    const tree = renderer.create(<Error message="message" buttonOnClick={mockCallBack}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('button callback is called on click', () => {
    const mockCallBack = jest.fn();
    const error = shallow((<Error message="message" buttonOnClick={mockCallBack} />));

    const button = error.find('button');
    button.simulate('click');

    expect(button.prop('disabled')).toBeFalsy();
    expect(button.text()).toBe('Click here to retry');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  test('disabled button doesn\'t call callback', () => {
    const mockCallBack = jest.fn();
    const label = "label";
    const error = shallow((<Error message="message" buttonOnClick={mockCallBack} buttonDisabled={true} buttonLabel={label} />));

    const button = error.find('button');
    button.simulate('click');

    expect(button.prop('disabled')).toBeTruthy();
    expect(button.text()).toBe(label);
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });

});
