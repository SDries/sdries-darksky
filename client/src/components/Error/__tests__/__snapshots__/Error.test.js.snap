// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-error renders with button 1`] = `
<div
  className="Error"
>
  <p>
    message
  </p>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Click here to retry
  </button>
</div>
`;

exports[`react-error renders with message 1`] = `
<div
  className="Error"
>
  <p>
    message
  </p>
</div>
`;
